libs/blocks/merch-card/merch-card.js

```diff
@@ -0,0 +1,204 @@
+/* eslint-disable prefer-destructuring */
+import { decorateButtons } from '../../utils/decorate.js';
+import { loadStyle, getConfig, createTag } from '../../utils/utils.js';
+import { addBackgroundImg, addWrapper, addFooter } from '../card/cardUtils.js';
+import { decorateLinkAnalytics } from '../../martech/attributes.js';
+import { replaceKey } from '../../features/placeholders.js';
+
+const { miloLibs, codeRoot } = getConfig();
+const base = miloLibs || codeRoot;
+
+const SEGMENT_BLADE = 'SegmentBlade';
+const SPECIAL_OFFERS = 'SpecialOffers';
+const PLANS_CARD = 'PlansCard';
+
+const getPodType = (styles) => {
+  const cardTypes = {
+    segment: SEGMENT_BLADE,
+    'special-offers': SPECIAL_OFFERS,
+    plans: PLANS_CARD,
+  };
+  const authoredType = styles?.find((style) => style in cardTypes);
+  return cardTypes[authoredType] || SEGMENT_BLADE;
+};
+
+const createDescription = (rows, cardType) => {
+  const descriptions = rows.slice(0, rows.length - 1);
+  const descriptionWrapper = createTag('div', { class: `consonant-${cardType}-description` });
+  descriptions?.forEach((description) => descriptionWrapper.appendChild(description));
+  return descriptionWrapper;
+};
+
+const createTitle = (titles, cardType) => {
+  const titleWrapper = createTag('div', { class: `consonant-${cardType}-title` });
+  titles?.forEach((title) => titleWrapper.appendChild(title));
+  return titleWrapper;
+};
+
+const decorateFooter = (el, styles, cardType) => {
+  const cardFooter = el.querySelector('.consonant-CardFooter');
+  const replacePlaceHolder = async (key, defaultValue) => {
+    const replacedKey = await replaceKey(key, getConfig(), defaultValue);
+    return { replacedKey } || 'Secure transaction';
```

Comment:
You're returning an object or a string, but on line 46 you always assume an object.

Reply:
Updated. Thanks accidentally added.